### PR TITLE
8271615: vmTestbase/vm/jit/LongTransitions/ test fail due to locale mismatch

### PR DESCRIPTION
**Pull request by mseledts**
Decimal point differs based on locale, could be '.' or ','.                                                                                                                          
Due to settings on a test host environment locale may differ between                                                                                                                 
the native side vs Java side thus causing mismatch and test failure.

Solution: added special handling in comparison loop for decimal points.

Testing: PASS (see bug comments for details)

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8271615](https://bugs.openjdk.org/browse/JDK-8271615): vmTestbase/vm/jit/LongTransitions/ test fail due to locale mismatch

### Reviewers
 * [Aleksey Shipilev](https://openjdk.org/census#shade) (@shipilev - **Reviewer**)
 * [Leonid Mesnik](https://openjdk.org/census#lmesnik) (@lmesnik - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10033/head:pull/10033` \
`$ git checkout pull/10033`

Update a local copy of the PR: \
`$ git checkout pull/10033` \
`$ git pull https://git.openjdk.org/jdk pull/10033/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10033`

View PR using the GUI difftool: \
`$ git pr show -t 10033`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10033.diff">https://git.openjdk.org/jdk/pull/10033.diff</a>

</details>

### CODE REVIEW

File: test/hotspot/jtreg/vmTestbase/vm/jit/LongTransitions/LTTest.java [hunk 1]
```diff
@@ -6218,6 +6218,15 @@ private static boolean chkFile()
...
             {
+                // Special handling for a decimal point.
+                // Decimal point differs based on locale, could be '.' or ','.
+                // Due to settings on a test host environment locale may differ between
+                // the native side vs Java side thus causing mismatch and test failure.
+                if ( cData[cnt] == '.' || cData[cnt] == ',' ) {
+                    if (javaData[cnt] == '.' || javaData[cnt] == ',') {
+                        continue;
+                    }
+                }
```
Comment (shipilev) [inline test/hotspot/jtreg/vmTestbase/vm/jit/LongTransitions/LTTest.java:6229]:
```suggestion
                if ((cData[cnt] == '.' || cData[cnt] == ',') &&
                    (javaData[cnt] == '.' || javaData[cnt] == ',')) {
                    continue;
                }
```

**Review by shipilev — COMMENTED** [review]
Ouch. I would have expected the default JVM locale inherited from the system locale. Did it start to fail due to "JEP 400: UTF-8 by Default"?

**Comment by mseledts** [comment]
Not sure about when this started failing. The bug was a round for a while, just getting to it now.

**Review by shipilev — APPROVED** [review]

**Review by lmesnik — APPROVED** [review]

**Comment by mseledts** [comment]
Thanks Aleksey, Leonid.

/integrate

**Pull request closed without merging**